cli/cmd/top.go

```diff
@@ -483,33 +492,70 @@ func processEvents(eventCh <-chan pb.TapEvent, requestCh chan<- topRequest, done
 	}
 }
 
-func pollInput(closing chan<- struct{}) {
+func pollInput(done chan<- struct{}, scrollLeft chan<- int, scrollRight chan<- int) {
 	for {
 		switch ev := termbox.PollEvent(); ev.Type {
 		case termbox.EventKey:
 			if ev.Ch == 'q' || ev.Key == termbox.KeyCtrlC {
-				closing <- struct{}{}
+				close(done)
 				return
 			}
+			if ev.Ch == 'a' || ev.Key==termbox.KeyArrowRight {
+				scrollLeft <- 1
+			}
+			if ev.Ch == 'd' || ev.Key==termbox.KeyArrowLeft  {
+				scrollRight <- 1
+			}
 		}
 	}
 }
 
-func renderTable(table *topTable, requestCh <-chan topRequest, done <-chan struct{}) {
-	ticker := time.NewTicker(100 * time.Millisecond)
-
+func renderTable(table *topTable, requestCh <-chan topRequest, done <-chan struct{}, scrollLeft <-chan int, scrollRight <-chan int) {
+	ticker := time.NewTicker(50 * time.Millisecond)
+	tablewidth:=0
+	scrollXpos := 0
+	width := 0
+	height := 0
+	width, height = termbox.Size()
+	tablewidth =table.adjustColumnWidths()
 	for {
+
 		select {
 		case <-done:
 			return
 		case req := <-requestCh:
 			table.insert(req)
 		case <-ticker.C:
-			termbox.Clear(termbox.ColorDefault, termbox.ColorDefault)
-			table.adjustColumnWidths()
-			table.renderHeaders()
-			table.renderBody()
-			termbox.Flush()
+			select {
+			case <-scrollLeft:
```

Comment:
I'd expect us to be reading off the scroll channels in the outer select rather than here.  That way we can accept scroll events and update the xPos at any time, instead of waiting for a render tick. As this is written, you can only scroll once per render frame, which is probably fine in practice since I don't think anyone is trying to scroll more than once every 50ms.  

But still, I think this would be cleaner if the code to update the scrollXpos was in the outer select and this timer select was solely dedicated to rendering.  This helps remove some of the code duplication here.

Reply:
@adleong Now I have created only one channel called `horizontalScroll`, which is of type chan int. I have also created the variable` xOffset` as mentioned. :)